### PR TITLE
[Feature] Add Inventory Slot IDs Vector Method for Loops

### PR DESCRIPTION
**Pull request by Kinglykrab**
# Description
- Adds `EQ::InventoryProfile::GetInventorySlotIDs()` which returns all inventory slots in a vector for easy looping.
- This allows us to reduce quite a bit of code duplication.

## Type of Change
- [X] New feature

# Checklist
- [X] I have tested my changes
- [X] I have performed a self-review of my code. Ensuring variables, functions and methods are named in a human-readable way, comments are added only where naming of variables, functions and methods can't give enough context.
- [X] I own the changes of my code and take responsibility for the potential issues that occur

### CODE REVIEW

File: common/inventory_profile.cpp [hunk 1]
```diff
@@ -1743,3 +1673,30 @@ std::vector<uint32> EQ::InventoryProfile::GetAugmentIDsBySlotID(int16 slot_id)
...
+		{ InventoryFilter::SharedBank, { EQ::invbag::SHARED_BANK_BAGS_BEGIN, EQ::invbag::SHARED_BANK_BAGS_END }},
+	};
+
+	std::vector<int16> slot_ids;
+
+	for (const auto& e : slots) {
+		if (filter == InventoryFilter::All || filter & e.first) {
+			for (int16 slot_id = e.second[0]; slot_id <= e.second[1]; ++slot_id) {
+				slot_ids.emplace_back(slot_id);
+				LogInfo("Adding slot ID [{}]", slot_id);
```
Comment (Akkadius) [inline common/inventory_profile.cpp:1696]: Probably don't want to leave this in here as LogInfo

File: common/inventory_profile.cpp [hunk 2]
```diff
@@ -1743,3 +1673,30 @@ std::vector<uint32> EQ::InventoryProfile::GetAugmentIDsBySlotID(int16 slot_id)
 
 	return augments;
 }
+
+const std::vector<int16>& EQ::InventoryProfile::GetInventorySlotIDs(uint8 filter)
+{
+	static const std::map<uint8, std::vector<int16>> slots = {
+		{ InventoryFilter::Worn,       { EQ::invslot::EQUIPMENT_BEGIN,       EQ::invslot::EQUIPMENT_END }},
+		{ InventoryFilter::Personal,   { EQ::invslot::GENERAL_BEGIN,         EQ::invslot::GENERAL_BEGIN }},
+		{ InventoryFilter::Personal,   { EQ::invbag::GENERAL_BAGS_BEGIN,     EQ::invbag::GENERAL_BAGS_BEGIN }},
```
Comment (joligario) [inline common/inventory_profile.cpp:1682]: GENERAL_BAGS_END

File: common/inventory_profile.cpp [hunk 3]
```diff
@@ -1743,3 +1673,30 @@ std::vector<uint32> EQ::InventoryProfile::GetAugmentIDsBySlotID(int16 slot_id)
...
+		{ InventoryFilter::Cursor,     { EQ::invbag::CURSOR_BAG_BEGIN,       EQ::invbag::CURSOR_BAG_END }},
+		{ InventoryFilter::Bank,       { EQ::invslot::BANK_BEGIN,            EQ::invslot::BANK_END }},
+		{ InventoryFilter::Bank,       { EQ::invbag::BANK_BAGS_BEGIN,        EQ::invbag::BANK_BAGS_END }},
+		{ InventoryFilter::SharedBank, { EQ::invslot::SHARED_BANK_BEGIN,     EQ::invslot::SHARED_BANK_END }},
+		{ InventoryFilter::SharedBank, { EQ::invbag::SHARED_BANK_BAGS_BEGIN, EQ::invbag::SHARED_BANK_BAGS_END }},
```
Comment (joligario) [inline common/inventory_profile.cpp:1687]: I know its forgiving, but we can remove final comma

File: zone/client.cpp [hunk 4]
```diff
@@ -10595,38 +10581,27 @@ uint32 Client::GetItemCooldown(uint32 item_id)
...
-						DeleteItemInInventory(slot_id, amount_left, true);
-					}
+	for (const int16& slot_id : EQ::InventoryProfile::GetInventorySlotIDs()) {
+		if (removed_count == quantity) {
+			break;
+		}
+
+		item = GetInv().GetItem(slot_id);
+		if (item && item->GetID() == item_id) {
+			int16 charges    = item->IsStackable() ? item->GetCharges() : 0;
```
Comment (joligario) [inline zone/client.cpp:10595]: Referenced above. I know you didn't write it, but we should be consistent with charges.

**Review by joligario — CHANGES_REQUESTED** [review]
A few comments. I think this deserves some good testing with a character with partial, full, and no inventory items across the spectrum. This should include stacks, charges, etc.

**Review by joligario — APPROVED** [review]
Looks good. A deliberate testing round is still warranted.

**Pull request closed without merging**